api/users/users.model.js

```diff
@@ -0,0 +1,55 @@
+const mongoose = require("mongoose");
+
+const userSchema = new mongoose.Schema(
+  {
+    email: {
+      type: String,
+      unique: true,
+      required: true,
+    },
+    password: {
+      type: String,
+    },
+    subscription: {
+      type: String,
+      enum: ["free", "pro", "premium"],
+      default: "free",
+    },
+    token: String,
+  },
+  { versionKey: false }
+);
+
+class User {
+  constructor() {
+    this.db = mongoose.model("User", userSchema);
+  }
+
+  getUsers = async (query) => {
+    const { sub: subscription } = query;
+    if (subscription) {
+      return await this.db.find({ subscription });
+    }

```

Comment:
this function can't be void 

this function should return array or null or throw error